library/src/test/java/io/getstream/chat/android/client/api/ClientConnectionTests.kt

```diff
@@ -78,7 +80,7 @@ internal class ClientConnectionTests {
     @Test
     fun successConnection() {
 
-        `when`(socket.events()).thenReturn(JustObservable(connectedEvent))
+        `when`(socket.events()).thenReturn(JustObservable(DisconnectedEvent(EventType.HEALTH_CHECK, Date())))
```

Comment:
Why is it a `DisconectedEvent`?? Shouldn't it be a `ConnectedEvent`? :thinking:

Reply:
IMO no. `events()` are called in `init()` and after receiving ConnectedEvent we are setting the state (e.g. user). We want to test if socket was connected after calling setUser() but it will only be called when there was no previous user

Reply:
Ok, I see.
The problem is the test configuration then.
We are testing `client.setUser()`, nothing else. We shouldn't know any "implementation details" about this object.
`Client` should be created out of the test (I am not sure if we can reuse the same instance, if not, it could be initialized into `before()` method) and mocks (on that case `socket`) reset

BTW, `DisconnectedEvent(EventType.HEALTH_CHECK)` is not coherent. if we need to create a `DisconnectedEvent` it should be `DisconnectedEvent(EventType.CONNECTION_DISCONNECTED, Date())`

Reply:
I agree with that and I think we should rewrite those tests and add more `ChatClientImpl` tests. I will slightly change `JustObservable` implementation to do requested changes but we should get back to this when we have more time.